### PR TITLE
Feature player page

### PR DESCRIPTION
**Pull request by Nilint1**
Player page initial page.

### CODE REVIEW

File: components/PlayerTable.tsx [hunk 1]
```diff
@@ -175,43 +183,41 @@ Props): JSX.Element {
             <TableBody {...getTableBodyProps()}>
               {hasData && pagination
                 ? page.map((row, i) => {
-                    prepareRow(row);
-
```
Comment (Nilint1) [inline components/PlayerTable.tsx:179]: just a removed line

File: components/PlayerTable.tsx [hunk 2]
```diff
@@ -175,43 +183,41 @@ Props): JSX.Element {
...
+                            {cell.render('Cell')}
+                          </td>
+                        );
+                      })}
+                    </tr>
+                  );
+                })
                 : rows.map((row, i) => {
-                    prepareRow(row);
-
```
Comment (Nilint1) [inline components/PlayerTable.tsx:198]: removed line again

File: components/RatingsTable/GoalieRatingsTable.tsx [hunk 3]
```diff
@@ -25,8 +27,23 @@ function GoalieRatingsTable({
         {
           Header: 'Player',
           id: 'player-table-player',
-          accessor: 'name',
+          accessor: ({ name, league, id }) => [
```
Comment (Nilint1) [inline components/RatingsTable/GoalieRatingsTable.tsx:30]: add a link to the player page

File: components/PlayerTable.tsx [hunk 4]
```diff
@@ -4,13 +4,15 @@ import styled from 'styled-components';
 
 // import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';
 // import Link from '../../components/LinkWithSeason';
-import { Player, Goalie, SearchType } from '../..';
```
Comment (Nilint1) [inline components/PlayerTable.tsx:7]: The main gist of all these changes in this file is that we had a lot of repeated code.  I generalized the ratings and stats table to use only one table that takes different parameters.  This makes sure all our tables only have to be written once if we want to make a change.

File: components/RatingsTable/GoalieRatingsTable.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 
 import { GoalieRatings } from '../..';
-
-import RatingsTable from '.';
+import Link from '../../components/LinkWithSeason';
+import PlayerTable from '../PlayerTable';
```
Comment (Nilint1) [inline components/RatingsTable/GoalieRatingsTable.tsx:5]: use new generic table

File: components/RatingsTable/SingleGoalieRatingsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
+import React from 'react';
```
Comment (Nilint1) [inline components/RatingsTable/SingleGoalieRatingsTable.tsx:1]: this is the component for a goalie page.

File: components/RatingsTable/SingleSkaterRatingsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,215 @@
+import React from 'react';
```
Comment (Nilint1) [inline components/RatingsTable/SingleSkaterRatingsTable.tsx:1]: skater page

File: components/RatingsTable/SkaterRatingsTable.tsx [hunk 8]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
 
-// import Link from '../../components/LinkWithSeason';
 import { PlayerRatings } from '../..';
-
-import RatingsTable from '.';
+import Link from '../../components/LinkWithSeason';
```
Comment (Nilint1) [inline components/RatingsTable/SkaterRatingsTable.tsx:4]: use generalized table.

File: components/RatingsTable/SkaterRatingsTable.tsx [hunk 9]
```diff
@@ -26,8 +26,23 @@ function PlayerRatingsTable({
...
+          accessor: ({ name, league, id }) => [
+            name,
+            league,
+            id,
+          ],
           // Create cell which contains link to player
+          Cell: ({ value }) => {
+            return (
+              <Link
+                href="/[league]/player/[id]"
```
Comment (Nilint1) [inline components/RatingsTable/SkaterRatingsTable.tsx:38]: link to player page

File: components/ScoreTable/GoalieScoreTable.tsx [hunk 10]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
 
-// import Link from '../../components/LinkWithSeason';
 import { Goalie } from '../..';
-
-import ScoreTable from '.';
+import Link from '../../components/LinkWithSeason';
```
Comment (Nilint1) [inline components/ScoreTable/GoalieScoreTable.tsx:4]: use generic table

File: components/ScoreTable/GoalieScoreTable.tsx [hunk 11]
```diff
@@ -26,8 +26,23 @@ function GoalieScoreTable({
...
+          accessor: ({ name, league, id }) => [
+            name,
+            league,
+            id,
+          ],
           // Create cell which contains link to player
+          Cell: ({ value }) => {
+            return (
+              <Link
+                href="/[league]/player/[id]"
```
Comment (Nilint1) [inline components/ScoreTable/GoalieScoreTable.tsx:38]: add link to player page

File: components/ScoreTable/SingleGoalieScoreTable.tsx [hunk 12]
```diff
@@ -0,0 +1,123 @@
+import React from 'react';
```
Comment (Nilint1) [inline components/ScoreTable/SingleGoalieScoreTable.tsx:1]: goalie stats

File: components/ScoreTable/SingleSkaterAdvStatsTable.tsx [hunk 13]
```diff
@@ -0,0 +1,158 @@
+import React from 'react';
```
Comment (Nilint1) [inline components/ScoreTable/SingleSkaterAdvStatsTable.tsx:1]: adv skater stat table

File: components/ScoreTable/SingleSkaterScoreTable.tsx [hunk 14]
```diff
@@ -0,0 +1,303 @@
+import React from 'react';
```
Comment (Nilint1) [inline components/ScoreTable/SingleSkaterScoreTable.tsx:1]: skater stat table

File: components/ScoreTable/SkaterAdvStatsTable.tsx [hunk 15]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
 
-// import Link from '../../components/LinkWithSeason';
 import { Player } from '../..';
-
-import ScoreTable from '.';
+import Link from '../../components/LinkWithSeason';
+import PlayerTable from '../PlayerTable';
```
Comment (Nilint1) [inline components/ScoreTable/SkaterAdvStatsTable.tsx:5]: use generic table

File: components/ScoreTable/SkaterAdvStatsTable.tsx [hunk 16]
```diff
@@ -26,8 +27,24 @@ function SkaterAdvStatsTable({
...
-          accessor: 'name',
           title: 'Player Name',
+          accessor: ({ name, league, id }) => [
+            name,
+            league,
+            id,
+          ],
+          // Create cell which contains link to player
+          Cell: ({ value }) => {
+            return (
```
Comment (Nilint1) [inline components/ScoreTable/SkaterAdvStatsTable.tsx:38]: add link to player page

File: components/ScoreTable/SkaterScoreTable.tsx [hunk 17]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
 
-// import Link from '../../components/LinkWithSeason';
 import { Player } from '../..';
-
-import ScoreTable from '.';
+import Link from '../../components/LinkWithSeason';
+import PlayerTable from '../PlayerTable';
```
Comment (Nilint1) [inline components/ScoreTable/SkaterScoreTable.tsx:5]: generic table

File: components/ScoreTable/SkaterScoreTable.tsx [hunk 18]
```diff
@@ -31,8 +31,23 @@ function SkaterScoreTable({
...
+            name,
+            league,
+            id,
+          ],
           // Create cell which contains link to player
+          Cell: ({ value }) => {
+            return (
+              <Link
+                href="/[league]/player/[id]"
+                as={`/${leagues[value[1]]}/player/${value[2]}`}
```
Comment (Nilint1) [inline components/ScoreTable/SkaterScoreTable.tsx:44]: link to player table

File: hooks/useGoalieInfo.ts [hunk 19]
```diff
@@ -0,0 +1,30 @@
+import useSWR from 'swr';
```
Comment (Nilint1) [inline hooks/useGoalieInfo.ts:1]: hook for single goalie

File: hooks/useGoalieRatingsId.ts [hunk 20]
```diff
@@ -0,0 +1,27 @@
+import useSWR from 'swr';
```
Comment (Nilint1) [inline hooks/useGoalieRatingsId.ts:1]: hook for single goalie

File: hooks/useGoalieStatsId.ts [hunk 21]
```diff
@@ -0,0 +1,31 @@
+import useSWR from 'swr';
```
Comment (Nilint1) [inline hooks/useGoalieStatsId.ts:1]: hook for single goalie

File: hooks/useRatingsId.ts [hunk 22]
```diff
@@ -0,0 +1,27 @@
+import useSWR from 'swr';
```
Comment (Nilint1) [inline hooks/useRatingsId.ts:1]: single ratings

File: hooks/useSkaterInfo.ts [hunk 23]
```diff
@@ -0,0 +1,30 @@
+import useSWR from 'swr';
```
Comment (Nilint1) [inline hooks/useSkaterInfo.ts:1]: single info

File: hooks/useSkaterStatsId.ts [hunk 24]
```diff
@@ -0,0 +1,31 @@
+import useSWR from 'swr';
```
Comment (Nilint1) [inline hooks/useSkaterStatsId.ts:1]: single stats

File: public/swagger.json [hunk 25]
```diff
@@ -476,6 +476,45 @@
         }
       }
     },
+    "/players/{id}": {
```
Comment (Nilint1) [inline public/swagger.json:479]: added api documentation

**Review by esilverm — APPROVED** [review]

**Pull request merged**